python/cudf/cudf/core/_internals/expressions.py

```diff
@@ -115,7 +115,7 @@ def visit_Name(self, node):
         self.stack.append(ColumnReference(col_id))
 
     def visit_Constant(self, node):
-        if not isinstance(node, ast.Num):
+        if not isinstance(node, ast.Num) and not isinstance(node, ast.Str):
```

Comment:
`isinstance` supports sequences:
```suggestion
        if not isinstance(node, (ast.Num, ast.Str)):
```